src/mesh/difops.cxx

```diff
@@ -168,15 +168,15 @@ const Field3D Grad_parP(const Field3D &apar, const Field3D &f) {
 *******************************************************************************/
 
 const Field2D Vpar_Grad_par(const Field2D &v, const Field2D &f, const CELL_LOC outloc) {
-  return v.getCoordinates(outloc)->Vpar_Grad_par(v, f, outloc);
+  return f.getCoordinates(outloc)->Vpar_Grad_par(v, f, outloc);
```

Comment:
I think this change (and two others that are the same) only makes a difference if `f` and `v` are on different meshes, which we don't support. Is there a particular motivation for this change (e.g. to be consistent with other places in the code)?

Reply:
Well, if they are staggered they are on the same mesh, but at different locations therein.

This worked before, and is used in e.g. STORM.

It also makes sense, as you want to advect quantities, then the advected quantity may ether be also staggered (e.g. velocity), or not (e.g. density).  In most cases, you would want the derivative at the advected quantity, which would be the one of `f`.

Sorry if I failed to address you comment, in which case I will try to clarify ...

Reply:
As we're passing `outloc` to `getCoordinates` in each case the location of the returned coordinates is entirely independent of which field we're using to fetch the coordinates from.

Reply:
Ah I see.
`outloc` can be `CELL_DEFAULT` - in which case it matters.

Reply:
Ah yes, good point I forgot about `CELL_DEFAULT`.